lib/cuckoo/core/database.py

```diff
@@ -957,7 +957,7 @@ def filter_machines_by_arch(machines, arch):
         Allow x64 machines to be returned when requesting x86.
         """
         if arch:
-            if arch == "x86":
+            if "x86" in arch:
                 # Prefer x86 machines over x64 if x86 is what was requested.
                 machines = machines.filter(Machine.arch.in_(("x64", "x86"))).order_by(Machine.arch.desc())
             else:
```

Comment:
question: Does this mean that, if the task has no arch tags, then it wouldn't find a machine at all because it would be looking for machines whose architecture is IN an empty list?

Reply:
hm it always adds x86 tag if no other matches any criteria

Reply:
It is possible for a task to have no arch tag if you have both of these config options switched off
https://github.com/kevoreilly/CAPEv2/blob/master/conf/web.conf#L41:L44

I have both switched off in my environment and tasks are able to get assigned.